examples/hackweek_demo/acquire_data.py

```diff
@@ -0,0 +1,130 @@
+    # turn to dataframes
+    pool_df = pd.DataFrame(pool).T
+    pool_df["shareReserves"] = pool_df["shareReserves"].astype(float)
+    pool_df["bondReserves"] = pool_df["bondReserves"].astype(float)
+    pool_df["lpTotalSupply"] = pool_df["lpTotalSupply"].astype(float)
+    pool_df["sharePrice"] = pool_df["sharePrice"].astype(float)
+    pool_df["timestamp"] = pool_df["timestamp"].iloc[0] + pool_df["blockNumber"] * 12
+    pool_df["spot_price"] = calculate_spot_price(
+        share_reserves=pool_df["shareReserves"],
+        bond_reserves=pool_df["bondReserves"],
+        lp_total_supply=pool_df["lpTotalSupply"],
+    )
+    txns, logs, receipts = [], [], []
+    for block in trades.values():
+        for txn in block:
+            txns.append(txn["transaction"])  # 1 per txn
+            logs.extend(
+                {k: v for k, v in log.items() if k != "args"} | log["args"]
+                for log in txn["logs"]
+                if log["event"] not in ["Approval", "Initialize"]
+            )
+            receipts.append(txn["receipt"])  # 1 per txn
+    print(f"{len(txns)=} {len(logs)=} {len(receipts)=}")
+    # merge pool_info fields
+    fields_to_add = ["blockNumber", "timestamp", "sharePrice", "shareReserves", "bondReserves", "lpTotalSupply"]
+    logs_df = pd.merge(pd.DataFrame(logs), pool_df.loc[:, fields_to_add])
+    return logs_df, pool_df
+
+
+def save_data(logs_df, pool_df, config_df, save_folder):
+    """Save data to csv."""
+    logs_df.to_csv(f"{save_folder}/logs.csv", index=False)
+    pool_df.to_csv(f"{save_folder}/pool.csv", index=False)
+    config_df.to_csv(f"{save_folder}/pool_config.csv", index=False)
+
+
+def main(
+    contracts_url: str,
+    ethereum_node: URI | str,
+    save_folder: str,
+    state_abi_file_path: str,
+    transactions_abi_file_path: str,
+    start_block: int,
+    first_deployed_block: int,
+    sleep_amount: float,
+):
+    """Main entry point for accessing contract & writing pool info"""
+    # pylint: disable=too-many-locals
+    # get web3 provider
+    web3_container: Web3 = contract_interface.setup_web3(ethereum_node)
+    # send a request to the local server to fetch the deployed contract addresses and
+    # load the deployed Hyperdrive contract addresses from the server response
+    state_hyperdrive_contract = contract_interface.get_hyperdrive_contract(
+        state_abi_file_path, contracts_url, web3_container
+    )
+    transactions_hyperdrive_contract = contract_interface.get_hyperdrive_contract(
+        transactions_abi_file_path, contracts_url, web3_container
+    )
+    # get pool config from hyperdrive contract
+    pool_config = contract_interface.get_smart_contract_read_call(state_hyperdrive_contract, "getPoolConfig")
+    pool_config["curve_fee"], pool_config["flat_fee"], pool_config["governance_fee"] = pool_config.pop("fees")
+    config_df = pd.DataFrame(pool_config, index=[0])
+    # initialize records
+    pool_info = {}
```

Comment:
I think there's a PoolInfo type lying around somewhere.  Maybe not if we removed it in favor of the db schema one...  We should have one though that works for the return value from hyperdrive, a TypeDict would work